AK/Array.h

```diff
@@ -35,8 +35,8 @@ struct Array {
     [[nodiscard]] constexpr T const& front() const { return at(0); }
     [[nodiscard]] constexpr T& front() { return at(0); }
 
-    [[nodiscard]] constexpr T const& back() const { return at(max(1, size()) - 1); }
-    [[nodiscard]] constexpr T& back() { return at(max(1, size()) - 1); }
+    [[nodiscard]] constexpr T const& back() const { return at(AK::max(1, size()) - 1); }
+    [[nodiscard]] constexpr T& back() { return at(AK::max(1, size()) - 1); }
```

Comment:
I don't remember seeing this, but - what?
`Array<T, 0>` is already weird, but `Array<T, 0>::back()` ends up accessing the 0th element too?
could we instead:
```suggestion
    [[nodiscard]] constexpr T& back() requires(size() > 0) { return at(size() - 1); }
```
or am I missing something here?

Reply:
Good point - that's a much better fix.